### PR TITLE
upgrades: InnoDB upgrade fix

### PR DESCRIPTION
**Pull request by egabancho**
- FIX Runs the sql statement which actually changes the engine from
  MyISAM to InnoDB.

Signed-off-by: Esteban J. G. Gabancho esteban.gabancho@gmail.com

### CODE REVIEW

File: invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py [hunk 1]
```diff
@@ -39,13 +39,14 @@ def do_upgrade():
...
     if current_app.config.get('CFG_DATABASE_TYPE') == 'mysql':
-        run_sql(
+        rows = run_sql(
             "SELECT CONCAT('ALTER TABLE ',TABLE_NAME,' ENGINE=InnoDB;')"
             " FROM INFORMATION_SCHEMA.TABLES"
             " WHERE ENGINE='MyISAM'"
             " AND table_schema = %s",
             (current_app.config.get('CFG_DATABASE_NAME'),)
         )
+        [run_sql(row[0]) for row in rows]
```
Comment (jirikuncar) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py]:
``` python
for row in rows:
    run_sql(row[0])
```

?

File: invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py [hunk 2]
```diff
@@ -39,13 +39,14 @@ def do_upgrade():
     """Carry out the upgrade."""
     from flask import current_app
     if current_app.config.get('CFG_DATABASE_TYPE') == 'mysql':
-        run_sql(
+        rows = run_sql(
             "SELECT CONCAT('ALTER TABLE ',TABLE_NAME,' ENGINE=InnoDB;')"
```
Comment (jirikuncar) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py]: `"SELECT TABLE_NAME"`

File: invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py]:
``` python
for row in rows:
    run_sql("ALTER TABLE '%s' ENGINE=InnoDB;", row[0])
```

Comment (egabancho) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py]: I'll try to do it using alembic instead, I'll update the PR asap

Comment (egabancho) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py]: It's not gonna happen :cry:

File: invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py [hunk 3]
```diff
@@ -39,13 +39,15 @@ def do_upgrade():
...
+        table_names = run_sql(
+            "SELECT TABLE_NAME"
             " FROM INFORMATION_SCHEMA.TABLES"
             " WHERE ENGINE='MyISAM'"
-            " AND table_schema = %s",
+            " AND table_schema=%s",
             (current_app.config.get('CFG_DATABASE_NAME'),)
         )
+        for table_name in table_names:
+            run_sql("ALTER TABLE `%s` ENGINE=InnoDB" % (table_name[0],))
```
Comment (jirikuncar) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py:50]:
```
`
```

vs.

```
'
```

## 

cc @hachreak

Comment (hachreak) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py:50]: it doesn't make sense to change because actually this query is valid only for mysql.

Comment (jirikuncar) [inline invenio_upgrader/upgrades/invenio_2015_07_14_innodb.py:50]: :+1:

**Comment by egabancho** [comment]
It seems to be running just fine for CDS database, although it takes a lot of time

**Comment by jirikuncar** [comment]
:shipit:

**Pull request merged**